### PR TITLE
Destroy modebar in purge step

### PR DESCRIPTION
**Pull request by mdtusz**
10 seconds of adding code,

30 minutes of hoping the jasmine test would finish without disconnecting...

From issue https://github.com/plotly/plotly.js/issues/120

### CODE REVIEW

**Comment by etpinard** [comment]
@mdtusz if you're up for it, this might a good to a `Plots.purge` jasmine test. 

- create a graph
- purge it
- check the gd that 'nothing' remains.

**Comment by mdtusz** [comment]
`purge` doesn't remove _all_ traces of the plot - `_context` is intentionally left. I suppose we can add a test to see if everything _except_ `_context` is removed.

**Comment by mdtusz** [comment]
@perigee the tests don't seem to be running properly on circle - I'm guessing this may be related to the changes you had made yesterday evening?

**Comment by etpinard** [comment]
@mdtusz I restarted the tests, and things work. CircleCI had a bad day yesterday.

File: test/jasmine/tests/plots_test.js [hunk 1]
```diff
@@ -298,4 +300,50 @@ describe('Test Plotly.Plots', function () {
...
+
+        it('should unset everything in the gd except _context', function () {
+            var expectedKeys = [
+                '_ev', 'on', 'once', 'removeListener', 'removeAllListeners',
+                'emit', '_context', '_replotPending', '_mouseDownTime',
+                '_hmpixcount', '_hmlumcount'
+            ];
+
+            Plotly.Plots.purge(gd);
+            expect(Object.keys(gd)).toEqual(expectedKeys);
```
Comment (etpinard) [inline test/jasmine/tests/plots_test.js]: @mdtusz great test :tada: thanks for doing this.

**Comment by etpinard** [comment]
10 seconds of adding code,

30 minutes of hoping the jasmine test would finish without disconnecting...

2 seconds to review.

:dancer:

**Pull request merged**